src/linear_algebra/quotient.lean

```diff
@@ -100,18 +101,57 @@ instance : add_comm_group (quotient p) :=
     refl },
   gsmul_neg' := by { rintros n ⟨x⟩, simp_rw [gsmul_neg_succ_of_nat, gsmul_coe_nat], refl }, }
 
-instance : has_scalar R (quotient p) :=
+section has_scalar
+
+variables {S : Type*} [has_scalar S R] [has_scalar S M] [is_scalar_tower S R M] (P : submodule R M)
+
+instance : has_scalar S (quotient P) :=
 ⟨λ a x, quotient.lift_on' x (λ x, mk (a • x)) $
- λ x y h, (quotient.eq p).2 $ by simpa [smul_sub] using smul_mem p a h⟩
+ λ x y h, (quotient.eq P).2 $ by simpa [smul_sub] using P.smul_mem (a • 1 : R) h⟩
+
+@[simp] theorem mk_smul (r : S) (x : M) : (mk (r • x) : quotient P) = r • mk x := rfl
+
+@[simp] theorem mk_nsmul (n : ℕ) : (mk (n • x) : quotient P) = n • mk x := rfl
+
+instance : is_scalar_tower S R P.quotient :=
+{ smul_assoc := by rintros x y ⟨z⟩; rw [quot_mk_eq_mk, ← mk_smul, smul_assoc, mk_smul, mk_smul] }
```

Comment:
More generally this should have a second type argument, and be stated as:
```lean
instance [is_scalar_tower S T R] : is_scalar_tower S T P.quotient :=
```
where `T` has all the same typeclass arguments as `S`.

I think if you do that the `is_scalar_tower S R (ideal.quotient I)` below falls out as the special case where `R = A = M`

Reply:
In fact, exactly the same proof works!